cmake/west.cmake

```
@@ -78,7 +78,7 @@ else()
     # possible to run 'python -m west.app.main'.
     string(APPEND WEST_MODULE .app.main)
   endif()
-  set(WEST ${PYTHON_EXECUTABLE} -m ${WEST_MODULE})
+  set(WEST ${PYTHON_EXECUTABLE} -m ${WEST_MODULE} CACHE FILEPATH "West")
```

If placed in cache, then it should be marked `INTERNAL`.
This is not a value users should control.

```suggestion
  set(WEST ${PYTHON_EXECUTABLE} -m ${WEST_MODULE} CACHE INTERNAL "West")
```

also line 55 should be updated, so that the value is placed in cache.

Also make comments around this code, in order to state why it is placed in cache.

---

Changed to `INTERNEL`.